crates/build-rs/src/input.rs

```diff
@@ -109,6 +109,10 @@ mod cfg {
     }
 
     /// If we are compiling with debug assertions enabled.
+    ///
+    /// This cfg is always true and misleading, so avoid setting it.
+    /// That is because Cargo queries rustc without any profile settings.
```

Comment:
```suggestion
    /// This cfg is always false and misleading, so avoid setting it.
    /// That is because Cargo explicitly filters out this setting since it queries rustc without any profile settings, and historically led to confusion.
```

Correct? The cfg is [filtered](https://github.com/rust-lang/cargo/blob/a4a3651d662c44e23827c8b42107844873f108df/src/cargo/core/compiler/custom_build.rs#L351-L355) so it is always unset.

Reply:
Yeah, copying it directly can be misleading.  I'm wanting to keep the comment in line with where its filtered, so I tweaked it in a different way.